pkg/trace/api/api.go

```diff
@@ -260,8 +297,13 @@ func (r *HTTPReceiver) Stop() error {
 	return nil
 }
 
+// headerResponseVersion is a response header sent to each request, representing
+// the trace-agent version.
+const headerResponseVersion = "Datadog-Trace-Agent-Version"
+
 func (r *HTTPReceiver) handleWithVersion(v Version, f func(Version, http.ResponseWriter, *http.Request)) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Add(headerResponseVersion, info.Version)
```

Comment:
@tylerbenson is this what you were thinking?

Reply:
Could you give an example of what that header would look like?  I'm wondering if it should be a list of endpoints, not just the current version.  That would enable a broader range of negotiation.

Reply:
It would technically be the version, i.e. `7.2.0`. I do remember you mentioning a list of endpoints, but what would that look like? And what would that mean to you? There may be other things that would be of interest to tracers in the future, that the version would reveal and a list of endpoints not.

How did you imagine this to work? A list of endpoints wouldn't say much about what each of them expects. If it does make sense and we want to do this, I would prefer we do it as part of another PR because it needs a small refactor: I would like the code that registers the endpoints to use the same source for those endpoints as would this response header. Otherwise people might end up adding to one but forgetting about the other.

I think having this header (the version) will be helpful nevertheless, and it's one small change I'm happy to keep here. 

Thoughts?

Reply:
ok, I'm fine having version included, but I worry that will imply a need to parse and do version number comparisons.

I was suggesting to have a header that looks like this `Trace-Agent-Endpoints: 0.3,0.4,0.5,1.0` such that if the tracer only supports up to `0.5` it would find that in the list and use it. Alternately, we could treat `0.3`/`0.4` as implied and not include them in the list.

Reply:
Hmm. True. You would have to parse, perhaps to a larger extent than just the versions. I'm afraid some versions might even look like `7.21.0-py3-jmx` or something. I think it's good to have this header, but perhaps it's not the best solution to what you're looking for.

I'm fine with your suggestion and happy to add it (in a follow-up PR), but I wonder if this is the best way. I'd be curious how other APIs and projects approach this problem. Surely it happens elsewhere too. 

What about [OPTIONS](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS#:~:text=The%20HTTP%20OPTIONS%20method%20is,refer%20to%20the%20entire%20server)? Maybe this is the "right" way to do it? It could be a JSON with various information.

Reply:
I agree... separate PR is best.

Options looks like an interesting... option.  :trollface:   I think it could be used for enabling/disabling various features.  The [ACCEPT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept) header is another one we could consider.  Here's an interesting blog on the topic:
https://www.troyhunt.com/your-api-versioning-is-wrong-which-is/